src/InodeCache.cpp

```diff
@@ -416,7 +416,12 @@ InodeCache::initialize()
         LOG("fstatfs failed: {}", strerror(errno));
         return false;
       }
-      if (buf.f_bavail * 512 < k_min_fs_mib_left * 1024 * 1024) {
+#if __FREEBSD__
+      if (static_cast<unsigned long>(buf.f_bavail) * 512
+          < k_min_fs_mib_left * 1024 * 1024) {
+#else
+      if ((buf.f_bavail) * 512 < k_min_fs_mib_left * 1024 * 1024) {
+#endif
```

Comment:
How about always casting to `uint64_t` regardless of platform? That way we don't need to duplicate the code.

Reply:
If that does not create problems on other platforms, it would be a great solution as we could avoid the extra include and the conditional compilation.

Reply:
I have updated the PR to do what you suggested and rebased it on master.